### DISCUSSION TITLE
Add attributes while using autoconfigured javaagent

### DISCUSSION THREAD

**Discussion by mindscratch** [open-telemetry/opentelemetry-java-instrumentation#4549] [Q&A]
I like the idea of using the autoconfigured JavaAgent to support tracing in my gRPC service, but I was wondering if it's possible to add attributes or baggage to the spans somehow? Perhaps a mix of autoconfigured + manual? Or would I have to just go all manual?

**Comment by trask** [comment]
hi @mindscratch! check out https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/manual-instrumentation.md#manual-instrumentation

**Reply by mindscratch** [reply]
@trask Thanks, that seems to work for the one method that processes the incoming request, but then I invoke a auto-generated gRPC client to call another service..and that span doesn't have the attribute. Is it possible to force the attribute to propagate through? Maybe a custom interceptor?

**Reply by trask** [reply]
oh yes, adding custom attributes to CLIENT spans is tricky in general. it's definitely worth trying a custom interceptor. though i'm not certain we are managing the interceptor order correctly, so it may work now and not later if we make a change: #2923

the general mechanism for adding custom attributes to CLIENT spans involves putting the attribute into the `Context` and then using a `SpanProcessor` to populate the attributes from the Context.

if you want to copy the attribute from the parent span down to the CLIENT span, you can skip putting the attribute into the `Context` and just write a `SpanProcessor` to copy it down, e.g. https://github.com/microsoft/ApplicationInsights-Java/blob/main/agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/InheritedAttributesSpanProcessor.java

once you get into SpanProcessors, you will have to write an agent extension: https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension#introduction

in the future, hopefully a more standard mechanism for this will evolve, maybe using Baggage and stamping baggage onto spans
